ledger-core/virtual/testutils/utils.go

```diff
@@ -6,9 +6,42 @@
 package testutils
 
 import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
 	"github.com/insolar/assured-ledger/ledger-core/vanilla/reflectkit"
 )
 
 func CmpStateFuncs(want, got interface{}) bool {
 	return reflectkit.CodeOf(want) == reflectkit.CodeOf(got)
 }
+
+// WaitSignalsTimed waits with timeout for signal channels to be closed
+// or signaled with a message once. Waiting is ordered and not parallel,
+// this may be beneficial in some situation and hurtful in others.
+func WaitSignalsTimed(t *testing.T, timeout time.Duration, signals ...<-chan struct{}) {
+	t.Helper()
+
+	cleanUp := make(chan struct{})
+	defer close(cleanUp)
+
+	allGood := make(chan struct{})
+	go func() {
+		defer close(allGood)
+
+		for _, signal := range signals {
+			select {
+			case <-signal:
+			case <-cleanUp:
+			}
+		}
+	}()
+
+	select {
+	case <-allGood:
+	case <-time.After(timeout):
+		require.FailNow(t, "timeout")
+	}
+}
```

Comment:
```suggestion
func WaitSignalsTimed(t *testing.T, timeout time.Duration, signals ...<-chan struct{}) {
	cleanUp := make(chan struct{})
	defer close(cleanUp)

	timeoutChannel := time.After(timeout)

	for _, signal := range signals {
		select {
		case <-signal:
		case <-timeoutChannel:
			require.FailNow(t, "timeout")
		}
	}
}
```

Parallel waiting is completely unnecessary in that case, btw. And it simplifies code

Reply:
switched to sequential reading of channels and it does simplify code. However, in a few situations this may result in a deadlock. Deadlock may happen when signal channels are not buffered, signaled with message instead of `close()` and two channels are interconnected. First channel in the list waits second to proceed after signaling, but second is blocked on writing into channel. It's so rare case that I decided to go with simpler code.